datafusion/sql/src/select.rs

```diff
@@ -384,7 +384,11 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     &[&[plan.schema()]],
                     &plan.using_columns()?,
                 )?;
-                let expr = col.alias(self.normalizer.normalize(alias));
+                let name = self.normalizer.normalize(alias);
+                let expr = match &col {
```

Comment:
Can you please add a comment here about what this is doing? I think it is avoiding adding an alias if the column name is the same. 

Do we have to worry about relation names here too (like what if `column.relation` is non null?)

Reply:
I understand, but I didn't expect the scenario where `column.relation` is inconsistent🤔

Reply:
I was thinking of a query like

```
EXPLAIN select a as a FROM table1 CROSS JOIN table2 order by a
```

Where both table1 and table2 have a column named `a`. I think we should add a test for this case too -- I'll try and do so later today

I would expect the test to fail

Reply:
as you expected
```sql
DataFusion CLI v33.0.0
❯ CREATE TABLE table1 (a int) as values(1);
0 rows in set. Query took 0.037 seconds.

❯ CREATE TABLE table2 (a int) as values(1);
0 rows in set. Query took 0.012 seconds.

❯ EXPLAIN select a as a FROM table1 CROSS JOIN table2 order by a;
Schema error: Ambiguous reference to unqualified field a
```